lib/puppet/type/ec2_vpc_vpn.rb

```diff
@@ -0,0 +1,62 @@
+Puppet::Type.newtype(:ec2_vpc_vpn) do
+  @doc = 'Type representing an AWS Virtual Private Networks.'
+
+  ensurable
+
+  newparam(:name, namevar: true) do
+    desc 'The name of the VPN.'
+    validate do |value|
+      fail 'VPNs must have a name' if value == ''
+    end
+  end
+
+  newproperty(:vpn_gateway) do
+    desc 'The VPN gateway to attach to the VPN.'
+  end
+
+  newproperty(:customer_gateway) do
+    desc 'The customer gateway to attach to the VPN.'
+  end
+
+  newproperty(:type) do
+    desc 'The type of VPN gateway, defaults to ipsec.1.'
+    defaultto 'ipsec.1'
+    validate do |value|
+      unless value =~ /^ipsec\.1$/
```

Comment:
I've replicated the API here quite closely (http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpnConnection.html), which allows you to pass a value here but currently only allows the single value. This to me suggested that might change at some point in the future, so allowing this to be explicit made sense.